dev/com.ibm.ws.microprofile.reactive.streams.operators_fat/fat/src/com/ibm/ws/microprofile/reactive/streams/test/suite/ReactiveConcurrentWorkTest.java

```diff
@@ -0,0 +1,122 @@
+ * <li> Application packaging is done in the @BeforeClass, instead of ant scripting.
+ * <li> Injects servers via @Server annotation. Annotation value corresponds to the
+ * server directory name in 'publish/servers/%annotation_value%' where ports get
+ * assigned to the LibertyServer instance when the 'testports.properties' does not
+ * get used.
+ * <li> Specifies an @RunWith(FATRunner.class) annotation. Traditionally this has been
+ * added to bytecode automatically by ant.
+ * <li> Uses the @TestServlet annotation to define test servlets. Notice that no @Test
+ * methods are defined in this class. All of the @Test methods are defined on the test
+ * servlet referenced by the annotation, and will be run whenever this test class runs.
+ */
+@RunWith(FATRunner.class)
+public class ReactiveConcurrentWorkTest extends FATServletClient {
+
+    public static final String APP_NAME = "ReactiveConcurrentWorkTest";
+
+    @Server("ReactiveStreamsTestServer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Create a WebArchive that will have the file name APP_NAME.war once it's written to a file
+        // Include the 'APP_NAME.web' package and all of it's java classes and sub-packages
+        // Automatically includes resources under 'test-applications/APP_NAME/resources/' folder
+        // Exports the resulting application to the ${server.config.dir}/apps/ directory
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addPackages(true, "com.ibm.ws.microprofile.reactive.streams.test.concurrent");
+
+        ShrinkHelper.exportDropinAppToServer(server, war);
+
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.stopServer();
+    }
+
+    /**
+     * Issue a slightly larger burst of work to Liberty
+     * this is used to enable observation of how worker threads
+     * are used.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testReactiveParallelClient() throws Exception {
+
+        int count = 3 * Runtime.getRuntime().availableProcessors();
+        final ExecutorService executor = Executors.newFixedThreadPool(count);
+
+        System.out.println(" Going to run " + count + " in parallel");
+
+        Collection<Callable<String>> tasks = new ArrayList<>();
+
+        for (int i = 0; i < count; i++) {
+            Callable<String> callableTask = () -> {
+                HttpUtils.findStringInReadyUrl(server, "/ReactiveConcurrentWorkTest/parallel?message=parallel" + count, FATServletClient.SUCCESS);
+                String msg = "Run on thread " + Thread.currentThread().getName();
+                System.out.println(msg);
+                return msg;
+            };
+            tasks.add(callableTask);
+        }
+
+        try {
+            List<Future<String>> futures = executor.invokeAll(tasks);
```

Comment:
I think you need to wait for these futures to complete (or timeout) before shutting down the executor and ending the test.

Reply:
OK I have inserted a wait for each (with a 5 min timeout)